paasta_tools/autoscaling/autoscaling_service_lib.py

```diff
@@ -355,6 +365,80 @@ def get_error_from_utilization(utilization, setpoint, current_instances):
         return 0.0
 
 
+def get_autoscaling_info(marathon_client, service, instance, cluster, soa_dir):
+    service_config = load_marathon_service_config(
+        service=service,
+        instance=instance,
+        cluster=cluster,
+        soa_dir=soa_dir,
+    )
+    all_marathon_tasks, all_mesos_tasks = get_all_marathon_mesos_tasks(marathon_client)
```

Comment:
This can be expensive, can you short-cirtuit this and only run it after line 376?